edx_exams/apps/lti/tests/test_views.py

```diff
@@ -1,20 +1,248 @@
+            },
+            'resource_link': {
+                'id': self.exam.resource_id
+            },
+            'attempt_number': self.attempt.attempt_number,
+            'action': 'flag',
+            'incident_time': '2018-02-01T10:45:33Z',
+            'incident_severity': '0.1',
+            'reason_code': '12056',
+            'reason_msg': 'Excessive background noise outside candidate control'
+        }
+
+        # Even though the client.post function below uses json.dumps to serialize the request as json,
+        # The json serialization needs to happen before the request for an unkown reason
+        request_body = json.dumps(request_body)
+        response = self.client.post(self.url, data=request_body, content_type='application/json',
+                                    HTTP_AUTHORIZATION='Bearer {}'.format(token))
+
+        if response.status_code == 200:
+            expected_msg = (
+                f'Flagging exam for user with id {self.user.anonymous_user_id} '
+                f'with resource id {self.exam.resource_id} and attempt number {self.attempt.attempt_number} '
+                f'for lti config id {self.test_provider.lti_configuration_id}, exam id {self.exam.id}, '
+                f'and attempt id {self.attempt.id}.'
+            )
+        else:
+            expected_msg = (
+                f'Attempt cannot be flagged for user with anonymous id {self.user.anonymous_user_id} '
+                f'with resource id {self.exam.resource_id} and attempt number {self.attempt.attempt_number} '
+                f'for lti config id {self.test_provider.lti_configuration_id}, exam id {self.exam.id}, and '
+                f'attempt id {self.attempt.id}. It has either not started yet, been rejcected, expired, or '
+                f'already verified.'
+            )
+
+        self.assertEqual(response.status_code, expected_response_status)
+        self.assertEqual(response.data, expected_msg)
+        mock_get_attempt.assert_called_once_with(self.user.id, self.attempt.attempt_number, self.exam.resource_id)
+
+    # TEST 3: get_attempt returned None -> Expected log
+    @ patch.object(Lti1p3ApiAuthentication, 'authenticate', return_value=(AnonymousUser(), None))
+    @ patch('edx_exams.apps.lti.views.LtiProctoringAcsPermissions.has_permission')
+    @ patch('edx_exams.apps.lti.views.get_attempt_for_user_with_attempt_number_and_resource_id')
+    def test_acs_no_attempt_found(self,
+                                  mock_get_attempt,
+                                  mock_permissions,
+                                  mock_authentication):  # pylint: disable=unused-argument
+        false_attempt_number = '88888888'
+
+        mock_get_attempt.return_value = None
+        mock_permissions.return_value = True
+
+        token = self._make_access_token('https://purl.imsglobal.org/spec/lti-ap/scope/control.all')
+
+        # Request w/ attempt number for an attempt that does not exist
+        request_body = {
```

Comment:
setting up this request body and making the request is the same across both of our tests here. Maybe a helper function to do this? Especially if we're adding another test for auth.